%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% This is just an example/guide for you to refer to when submitting manuscripts to Frontiers, it is not mandatory to use Frontiers .cls files nor frontiers.tex  %
% This will only generate the Manuscript, the final article will be typeset by Frontiers after acceptance.   
%                                              %
%                                                                                                                                                         %
% When submitting your files, remember to upload this *tex file, the pdf generated with it, the *bib file (if bibliography is not within the *tex) and all the figures.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% Version 3.4 Generated 2022/06/14 %%%
%%% You will need to have the following packages installed: datetime, fmtcount, etoolbox, fcprefix, which are normally inlcuded in WinEdt. %%%
%%% In http://www.ctan.org/ you can find the packages and how to install them, if necessary. %%%
%%%  NB logo1.jpg is required in the path in order to correctly compile front page header %%%

\documentclass[utf8]{FrontiersinHarvard} % for articles in journals using the Harvard Referencing Style (Author-Date), for Frontiers Reference Styles by Journal: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal
%\documentclass[utf8]{FrontiersinVancouver} % for articles in journals using the Vancouver Reference Style (Numbered), for Frontiers Reference Styles by Journal: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal
%\documentclass[utf8]{frontiersinFPHY_FAMS} % Vancouver Reference Style (Numbered) for articles in the journals "Frontiers in Physics" and "Frontiers in Applied Mathematics and Statistics" 

%\setcitestyle{square} % for articles in the journals "Frontiers in Physics" and "Frontiers in Applied Mathematics and Statistics" 
\usepackage{url,hyperref,lineno,microtype,subcaption}
\usepackage{svg}
\usepackage[onehalfspacing]{setspace}
\usepackage{listings}
\usepackage{rotating}
\usepackage[english]{babel}


\lstset{
  basicstyle=\ttfamily,
  breaklines=true,  % Enable line breaking
  postbreak=\mbox{$\hookrightarrow$}  % Symbol for breakline
}

\lstdefinestyle{BashInputStyle}{
  language=bash,
  basicstyle=\small\sffamily,
  numbers=left,
  numberstyle=\tiny,
  numbersep=5pt,
  frame=single,
  columns=fullflexible,
  backgroundcolor=\color{lightgray},
  linewidth=\linewidth,
  xleftmargin=0.075\linewidth,
  literate={-}{-}1 {'}{{\textquotesingle}}1,
}

\linenumbers


% Leave a blank line between paragraphs instead of using \\


\def\keyFont{\fontsize{8}{11}\helveticabold }
\def\firstAuthorLast{Weaver {et~al.}} %use et al only if is more than 1 author
\def\Authors{Steven Weaver\,$^{1*}$, Vanessa Davila Conn\,$^{2}$, Hannah Verdonk\,$^{1}$, Joel Wertheim\,$^{3}$, and Sergei L. Kosakovsky Pond\,$^{1}$}
% Affiliations should be keyed to the author's name with superscript numbers and be listed as follows: Laboratory, Institute, Department, Organization, City, State abbreviation (USA, Canada, Australia), and Country (without detailed address information such as city zip codes or street names).
% If one of the authors has a change of address, list the new address below the correspondence details using a superscript symbol and use the same symbol to indicate the author in the author list.
\def\Address{$^{1}$ Center for Viral Evolution, Temple University, Philadelphia, PA, USA \\
$^{2}$ Center for Research in Infectious Diseases, National Institute of Respiratory Diseases, Mexico City, Mexico  \\ 
$^{3}$ Department of Medicine, University of California, San Diego, CA
}
% The Corresponding Author should be marked with an asterisk
% Provide the exact contact address (this time including street name and city zip code) and email of the corresponding author
\def\corrAuthor{Steven Weaver}

\def\corrEmail{sweaver@temple.edu}

\begin{document}
\onecolumn
\firstpage{1}

\title { AUTO-TUNE: FINDING AN OPTIMAL DISTANCE THRESHOLD FOR INFERRING HIV TRANSMISSION CLUSTERS } 

\author[\firstAuthorLast ]{\Authors} %This field will be automatically populated
\address{} %This field will be automatically populated
\correspondance{} %This field will be automatically populated

\extraAuth{}% If there are more than 1 corresponding author, comment this line and uncomment the next one.
%\extraAuth{corresponding Author2 \\ Laboratory X2, Institute X2, Department X2, Organization X2, Street X2, City X2 , State XX2 (only USA, Canada and Australia), Zip Code2, X2 Country X2, email2@uni2.edu}


\maketitle


\begin{abstract}

%%% Leave the Abstract empty if your article does not require one, please see the Summary Table for full details.
\section{}

\noindent Choosing an appropriate distance threshold is an important part of inferring a transmission network to determine the relative growth of clusters within a localized epidemic. This distance threshold determines how close two consensus sequences must be in order for a link to be created between them in the network. Using a distance threshold that is too high can result in a network with many unnecessary links, making it difficult to interpret and analyze. On the other hand, using a distance threshold that is too low can result in a network with too few links, which may not capture key insights into rapidly growing clusters among patients with shared attributes that could benefit from public health intervention measures. 

Here, we present a heuristic scoring approach for tuning a distance threshold by associating each tested threshold against the maximal number of clusters created across all thresholds and the difference between the ratio ($R_{12}$) of the largest cluster in the network to the second largest cluster at each iteration. The number of clusters is normalized between $[0,1]$ then gated via a Gompertz function transform. Meanwhile, the distribution of all $R_{12}$ ratios are converted to $Z$ scores, and normalized relative to the largest positive $Z$ score across all candidate distances. The priority score is the sum of aforementioned two components. 

Published research using the HIV-TRACE software package frequently use the default threshold of $1.5\%$ for HIV pol gene sequences. We apply our scoring heuristic to outbreaks with different characteristics, such as regional or temporal variability, and demonstrate the utility of using the scoring mechanism's suggested distance threshold to identify clusters exhibiting risk factors that would have otherwise been more difficult to identify. For example, while we found that a $1.5\%$ distance threshold is typical for US-like epidemics, recent outbreaks like the CRF07\_BC subtype among men who have sex with men (MSM) in China has been found to have a lower optimal threshold of $0.5\%$ to better capture the transition from injected drug use (IDU) to MSM as the primary risk factor. Alternatively, in communities surrounding Lake Victoria, where there has been sustained transmission for several years, we found that a larger distance threshold is suitable to capture a more risk factor diverse populace with sparse sampling over a longer period of time. Such identification may allow for more informed intervention action by respective public health officials. 

\tiny
  \keyFont{ \section{Keywords:} molecular epidemiology, HIV, network, transmission cluster, surveillance} %All article types: you may provide up to 8 keywords; at least 5 are mandatory.
\end{abstract}

\section{Introduction}

Choosing an appropriate distance threshold is an important part of using a transmission network to track the spread of a contagious disease. This distance threshold determines how close two individuals must be in order for a link to be created between them in the network.

Using a distance threshold that is too small can result in a network with many unnecessary links, making it difficult to interpret and analyze. On the other hand, using a distance threshold that is too large can result in a network with too few links, making it difficult to accurately track the spread of the disease.

To ensure that the transmission network is useful and informative, it is important to carefully consider the appropriate distance threshold. This may vary depending on the specific disease and the context in which it is spreading. For example, a highly contagious respiratory illness may require a smaller distance threshold than a less contagious illness that is primarily spread through direct contact.

In general, the goal is to strike a balance between having enough links to accurately track the spread of the disease, while not having so many links that the network becomes difficult to interpret. This can be achieved through careful analysis and consideration of the specific disease and context.

Overall, choosing an appropriate distance threshold is an important step in using a transmission network to track the spread of a contagious disease. It can help ensure that the network is useful and informative, and can ultimately aid in efforts to control and prevent the spread of the disease.

\section{Methods}

\subsection{Scoring Heuristic Procedure}

Network threshold selection procedure proceeds as follows:

\begin{enumerate}
  \item{For each candidate threshold $d_L$, in increasing order, ranging from the smallest genetic distance in the dataset, up to either the largest distance or a predetermined maximal threshold, we compute two network statistics: $R_{12}$, the ratio of the largest cluster to the second largest cluster, and $C$ – the number of clusters in the network.}

  \item{ A priority score is assigned to each $d_L$. This score measures two properties of the threshold: Does $R_{12}$ jump at $d_L$? How far is the number of clusters $C$ at $d_L$ from the maximal number of clusters over all threshold values? Let there be $N$ overall $d_L$ candidate values, and assume we are examining the ith candidate, $d_L^i$ with $W < i \leq N - W$ ($W$ is a positive integer defined below).

  \begin{enumerate}
    \item{The $R_{12}$ jump is computed by looking at the normalized ratio of the mean $R_{12}$ values computed over the leading window $d_L^{i+1}…d_L^{i+W}$ and the trailing window $d_L^{i-W}… d_L^{i-1}$. The width of the window, $W$, is defined as $(([\frac{N}{100}],3),30)$. The distribution of ratios is converted to $Z$ scores, and normalized relative to the largest positive $Z$ score across all candidate distances, yielding the jump component of the score.}
    \item{The number of clusters, $C_i$ at threshold $d_L^i$ is first normalized to $[0,1]$ through $\frac{{C_{max} - C_i}}{{C_{max} -C_{min}}}$ and next gated via a Gompertz function transform ${1-e}^{-e^{-25x+3}}$ This function provides an ad hoc means for penalizing having too few clusters relative to the maximum over all ranges. For example, a threshold that yields $95\%$ of the maximal number of clusters receives a score of $0.996$, while a threshold that yields $85\%$ - a score of $0.376$.}
    \item{The priority score for $d_L^i$ is the sum of the two components defined in (a) and (b).}
  \end{enumerate}}


  \item{The threshold with the highest priority score will be selected as the suggested automatic distance threshold, if the score is high enough ($1.9$ or more), and either of the two conditions hold.
    \begin{enumerate}
      \item{No other thresholds have priority scores of $1.9$ or higher}
      \item{If other thresholds have priority scores of $1.9$ or higher, then the range of thresholds represented by these options is small (no more than $log(N)$ times the mean step between successive $d_L^i$).}
    \end{enumerate}}

 \item{If no single threshold can be selected in step 3, then the one with the highest priority score is suggested, and an inspection of the plot like the one on the analyze page is recommended to ensure that the threshold is sensible.}
\end{enumerate}

\subsection{Assortativity}

Degree-weighted homophily (DWH) is a measure of similarity between nodes in a network based on their attributes (such as demographic characteristics or behaviors) and their degree (i.e., the number of connections they have to other nodes in the network). It is used to quantify the extent to which nodes with similar attributes tend to be connected to each other more frequently than would be expected by chance.
DWH is calculated as the ratio of the observed number of connections between nodes with similar attributes to the expected number of connections between such nodes, based on their degree.

In mathematical terms, it is defined as:
\begin{equation}
  DWH = \frac{W_M + W_C - 2W_X}{\frac{d_{in}}{nodes_{in}^2} + \frac{d_{out}}{nodes_{out}^2}}
\end{equation}

Where
\begin{itemize}
  \item{$W_M$ : Weight of in-group connections}
  \item{$W_C$ : Weight of out-group connections}
  \item{$W_X$ : Weight of cross-group connections}
  \item{$d_{in}$ : In-group degree}
  \item{$d_{out}$ : Out-group degree}
  \item{$nodes_{in}$ : number of in-group nodes}
  \item{$nodes_{out}$ : number of out-group nodes}
\end{itemize}

DWH ranges from -1 to 1. A DWH value of 0 indicates that there is no more homophily than expected with chance, while a value of 1
indicates that there is perfect homophily (e.g. Birds always link to birds, and only birds). A value of -1 is achieved for perfectly disassortative networks (e.g. Bird never linking with another bird).

DWH is used in social network analysis and in the study of how different
attributes are related to the formation of connections between individuals. It
is used as a way to measure the similarity of attributes between individuals in
a network. Additionally, randomization is performed by shuffling attribute labels among nodes, then performing DWH computation. 
This is useful in creating a null distribution of DWH scores under random
mixing. A panmictic range is reported by shuffling attributes multiple times and
reporting the minimum and maximum score.

\subsection{Implementation}

The software implementation involves a step-by-step process that utilizes the HIV-TRACE suite of packages. It starts with calculating pairwise distances wth the tn93 tool and a supplied multiple sequence alignment. 
This generated pairwise distances are supplied to the hivnetworkcsv script while providing the -A keyword argument. A brief outline of the software's implementation are as follows

\begin{enumerate}

\item{ Calculate pairwise distances: The user first calculates the pairwise distances using the tn93 fast pairwise distance calculator, providing the necessary threshold value and the input FASTA file. The command for this step is 
\begin{lstlisting}[style=BashInputStyle]
 tn93 -t 0.030 pol.fasta > pairwise_distances.15.tn93.csv
\end{lstlisting}

Please note that the threshold should include the maximal range one is intending to test.

}

\item {Compute distance threshold scores: The hivnetworkcsv script is then executed with the required input file, format, and autotune option to generate a tab-separated output file, as shown below
\begin{lstlisting}[style=BashInputStyle]
 hivnetworkcsv -i pairwise_distances.15.tn93.csv -f plain -A 0 > autotune_report.tsv
\end{lstlisting}
}

\item {Visualize the report: Users can upload the generated autotune\_report.tsv file to \\
  \url{http://autotune.datamonkey.org/analyze} for visualization and further analysis of the data. This web-based platform provides an interactive environment to explore scores and other metrics across the range of tested outputs. }

\item {Run HIV-TRACE: Once AUTO-TUNEd threshold(s) are settled upon after review, the user runs the HIV-TRACE command with the appropriate input FASTA file, distance threshold, and other required arguments. The output is saved as a JSON file. An example command is
	\begin{lstlisting}[style=BashInputStyle]
hivtrace -i ./INPUT.FASTA -a resolve -r HXB2_prrt -t < autotune_threshold > -m 500 -g .05 > hivtrace.results.json
	\end{lstlisting}
}

\subsubsection{Optional : Compute Assortatviity Metrics}

\item{ Annotate results: The hivnetworkannotate script is used to annotate the results obtained from the HIV-TRACE step with attributes. The script takes the JSON results file, node attributes file, schema file, and a resolve flag as input.
	\begin{lstlisting}[style=BashInputStyle]
hivnetworkannotate -n hivtrace.results.json -a node_attributes.json -g schema.json -r
	\end{lstlisting}
 For more information, users can refer to the hivnetworkannotate documentation.
}

\item{ Analyze the results with DWH: After the results file has been annotated, the user can proceed to the assortativity page, http://autotune.datamonkey.org/assortativity, for further analysis of the output. }

AUTO-TUNE is readily accessible on GitHub as part of the hivclustering repository (https://github.com/veg/hivclustering). It is integrated into the command-line interface of the software as the -A or --auto-profile argument. hivclustering is a key component of the HIV-TRACE suite of tools, a resource for the inference, analysis and visualization of HIV transmission networks. 

The Degree Weighted Homophily (DWH) calculation tool, an integral component of the assortativity step, is developed using TypeScript, a statically typed superset of JavaScript that ensures robustness and scalability. In an effort to promote accessibility and ease of integration, the DWH tool is packaged and distributed through the Node Package Manager (NPM), enabling researchers and developers to conveniently incorporate this advanced analytical tool into their own projects and workflows. DWH can be used in-browser or as a command line tool, allowing researchers and developers to employ the tool in an interactive command-line interface or integrate it into larger software applications, thus catering to a diverse array of technical needs and preferences. Instructions for usage and installation is found on Github (https://github.com/veg/dwh).

\end{enumerate}

The described workflow offers a systematic approach to analyze potential distance thresholds for one's data with AUTO-TUNE, from calculating pairwise distances to visualizing and annotating results.

\subsection{Visualization}

Visualizations of AUTO-TUNE results are accessible at http://autotune.datamonkey.org/analyze. It is a dynamic and interactive web-based platform that offers visualization and analysis of results generated by AUTO-TUNE. 
The website provides a comprehensive view of the data by generating various plots across candidate distance thresholds. These include a score plot, allowing users to identify trends and anomalies across the full range of thresholds. Additionally, it generates a graph showing the number of clusters across candidate thresholds, one of the components that contribute to the score. The site also includes an R1/R2 plot that displays the ratio of the largest cluster to the second largest cluster across candidate thresholds, which is the other metric that contributes to the scoring heuristic. 

An assortativity tool is available at http://autotune.datamonkey.org/assortativity, and is an advanced analytical tool engineered to facilitate the calculation of Degree Weighted Homophily (DWH) values. It utilizes the DWH NPM package to generate a tabular representation of DWH values corresponding to each value for a selected attribute annotation, providing an exhaustive examination of the interrelationships for the field.
A notable feature is the computation of the panmictic range, which involves a label permutation test to generate the null distribution of DWH values. This feature establishes a comparative baseline that aids in determining the significance of homophily versus what would be expected by chance.
Lastly, the site also provides a plot of the fraction of pairwise connections, normalized by degree, for each value pertinent to the selected field. This visual depiction facilitates an intuitive comprehension of the distribution and interconnections within the dataset.

The site aims to offer a user-friendly interface for data visualization, playing an important role in interpreting and understanding AUTO-TUNE's output data. The visualization code is available on Github (https://github.com/stevenweaver/autotune-app/).


\subsection{Comparisons with previously published analyses}

In conducting our comparisons with the established clustuneR method, we
procured our datasets from \cite{wolf_short_2017} and
\cite{vrancken_multi-faceted_2017} utilizing the identical approach delineated
in \cite{chato_public_2020}. These datasets, namely Middle Tennessee,
Seattle, and Alberta, were processed using the workflow prescribed
in Section 2.3. This enabled us to determine an optimal threshold for each
dataset using our proposed method, AUTO-TUNE. We further executed the command
as detailed in step 4 of Section 2.3, deploying thresholds previously
established as optimal by \cite{chato_public_2020}.

To perform comparisons, we computed the average degree-weighted homophily score
over a set of three-year sliding windows. Specifically, the homophily among
nodes was calculated for a collection of date ranges as follows:

\begin{equation} \bar{H} = \frac{1}{N}\sum_{i=1}^{N} H(w_i) \end{equation}

where $\bar{H}$ represents the average degree-weighted homophily score, $N$
is the total number of sliding windows, $H(w_i)$ is the homophily score for
the $i$-th window, and the windows $w_i$ correspond to the date ranges, e.g.,
'2012-2015', '2013-2016', '2014-2017', etc. This methodology allowed us to
compare the "best thresholds" derived from our proposed AUTO-TUNE method
against those defined as optimal in \cite{chato_public_2020}.

Second, we set out to compare the thresholds obtained in original
investigations with those obtained by AUTO-TUNE. To select the data sets for
this analysis, we conducted a scientific literature search to identify studies
focused on HIV networks for public health purposes. We then filtered the
studies that utilized HIV-TRACE to infer genetic networks and had publicly
available sequences. Thus, we attempted to include studies from different
countries and regions, enabling us to assess the performance of our method
across various epidemic contexts, risk groups, and network sizes in real-data
sets that used variable clustering thresholds. 

In order to evaluate the influence of sampling density on the genetic distance
threshold as determined by AUTO-TUNE, we implemented a strategy of random
subsampling from the original dataset sourced from \cite{rhee_national_2019}.
This study was selected due to its satisfactory AUTO-TUNE score when utilized
in its entirety, as well as its inherent design as a Geographically-Stratified
set of 716 Pol Subtype/CRF (GSPS) reference sequence dataset. The dataset,
which comprises 6034 samples gathered between 1959 and 2016, was subjected to
random subsampling ten times at proportions of $25\%$, $50\%$, and $75\%$ of
the original sample size. For each subsample, the optimal threshold and
associated scores were determined via AUTO-TUNE.


\section{Results}

\subsection{Comparison with clustuneR}

We compared results to clustuneR, which employs the recency of sample
collection or diagnosis as individual-level weights in a predictive model to
estimate the growth of HIV clusters. The thresholds determined optimal by
clustuneR were found by finding the minimum GAIC (generalized AIC) across
candidate distances between $0$ and $0.04$ in steps of $8 \times 10^{-4}$. GAIC is
the difference between a null model that is only influenced by cluster size,
and a weighted model model that includes individual-level attributes among
known cases in the cluster. Using the minimum GAIC metric, it was found that
$0.016$ was the optimal threshold for Tennessee and Seattle, and $0.0104$ for
Northern Alberta.

In contrast, AUTO-TUNE does not incorporate any attribute data in its scoring
heuristic. Instead, it relies on clustering metrics constructed purely from
pairwise distances between sequences. Using the same datasets analyzed by
clustuneR \cite{chato_public_2020}, AUTO-TUNE found the thresholds with the
highest scores to be $0.01431$ for Middle Tennessee, $0.01354$ for Seattle, and
$0.01099$ for Northern Alberta. \autoref{tab:homophily} 

Performance of the inferred optimal thresholds were performed using an average
degree-weighted homophily (DWH) score across 3-year collection date windows
starting from the oldest collection year for each respective dataset, as that
is the metadata that was consistently available across all three datasets and
was the attribute of focus used by clustuneR. DWH in this case measures the
affinity for nodes within the network to link with other nodes in the same
collection date window. For example, samples collected between 2012-2015
linking more often with other samples within the same time window would result
in a higher DWH score. It was found that, across all three datasets, using the
threshold with the highest score reported by AUTO-TUNE resulted in a higher
average DWH score across all three datasets. 
% Add something about Wilcoxon test

When reviewing scores across all candidate thresholds with AUTO-TUNE
\label{fig:clustuner}. none of the three datasets reached one confident score
over any other. The most confident score was achieved by the Seattle dataset at
$1.53325$, then Tennessee with a high score of $1.25807$, and lastly Canada at
$1.01678$. When finding peaks, a second peak in Seattle denotes that $0.01166$
may also be an optimal threshold to consider, as its score, $1.52976$, is only
$0.003$ less than the highest score. The optimal score for Tennessee and Canada
are a bit more dubious, as there are multiple peaks within close scores of each
other. Indeed, after applying a $0.75$ minimum score threshold after visual
inspection for peak ranges, standard deviation among scores were $0.0475$,
$0.0089$, and $0.0084$ for Seattle, northern Alberta, and Tennessee,
respectively. This implies there may be multiple thresholds that would be
considered reasonable, and downstream homophily metrics with attributes may aid
in coming to a decision.

\subsection{Comparison with Prior Publications Citing HIV-TRACE}

% It is problematic to tune this threshold and most analyses use an ad-hoc threshold or evaluate sensitivity over a range of thresholds.

Next, we curated publications citing HIV-TRACE that also had publicly
accessible data associated with the study \cite{rhee_national_2019,
brenner_role_2021, h_acquisition_2021, liu_dynamics_2020, bbosa_short_2020,
yan_central_2020, dalai_combining_2018, sivay_hiv-1_2018}. We found that a
variety of different ways were used to determine distance thresholds, from
precedent set by the CDC for detecting recent and rapid clusters
\cite{yan_central_2020}, because they've been used before in other studies
\cite{sivay_hiv-1_2018}, to visual inspection of number of clusters and nodes
across candidate distance thresholds \cite{liu_dynamics_2020}. As a result, it
was found that since these thresholds are determined largely qualitatively, the
distance thresholds used tend to be round numbers and only somewhat tuned to
their respective geographic region of research. \autoref{tab:paperComparison}
When utilizing AUTO-TUNE for the same datasets, more granularity in selecting
thresholds can be performed in a straightforward manner. While it may seem
appealing to assume that being within a $0.5\%$ threshold of what AUTO-TUNE
considers optimal is good enough, further inspection into results reveals that
scores can vary widely within a very short distance. For example, when
reviewing the distribution of scores for the dataset used by
\cite{dalai_combining_2018}, the score is exactly $2$ at $0.01848$, but at
thresholds tested at just $0.00002$ difference, scores drop preciptously to
$1.638$ and $0.826$ for candidates $0.01846$ and $0.0185$, respectively.
Another example of a seemingly close threshold yet perhaps not optimal is found
with \cite{bbosa_short_2020}. While no score across candidate thresholds was
found to be above $1.9$, a high score was found at distance $1.707$, with
$1.2415$. Contrast this with the threshold used, $1.5\%$, with a score of
$0.0124$.

\subsection{Evaluating Performance with DWH}

To evaluate the performance of an AUTO-TUNEd optimized threshold using
degree-weighted homophily, we first evaluated a CRF07\_BC network with data
from China. $8178$ HIV-1 CRF07\_BC pol sequences were used from China to
construct longitudinal transmission networks, each pol sequence were annotated
with risk factor detailing whether the patient was heterosexual (Hetero),
Person With Injected Drug Use (PWID), or Men who have Sex with Men (MSM), among
other attributes. Using AUTO-TUNE, no distance threshold receives a score above
$1.9$, but using the default $1.5\%$ threshold is clearly suboptimal
\label{fig:chinaScorePlot}. Using a $1.5\%$ threshold, the network captures
$5923$ nodes, of which $559$ are PWID, $3371$ MSM, $1993$ Hetero, and received
an AUTO-TUNE score of $0.029$. When evaluating DWH among risk factors, MSM,
Hetero, and PWID had scores of $0.211$, $0.133$, and $0.168$, respectively. 

When using AUTO-TUNE, two separate ranges appear. The highest score is obtained
with distance threshold $0.76\%$ at $1.1369$. The second highest score is
$1.0303$ at distance threshold $0.0019$. Networks at both thresholds were also
evaluated with DWH based on risk factor. The network at $0.76\%$ captured
$3537$ nodes, of which $236$ are PWID, $2271$ MSM, $1030$ Hetero. When
evaluating DWH among risk factors, MSM and Hetero both had slightly increased
scores of $0.237$ and $0.185$, respectively. PWID DWH dramatically increased to
$0.401$. The network at $0.19\%$ captures $1654$ nodes, of which $151$ are
PWID, $1075$ MSM, $428$ Hetero. When evaluating DWH among risk factors, MSM,
Hetero, and PWID had slightly increased scores over $0.76\%$ of $0.292$,
$0.25$, and $0.445$, respectively.


\subsection{Effect on Sampling Density}




\section{Discussion}

AUTO-TUNE operates solely utilizing genetic sequence data to ascertain a
decisive threshold. It employs a scoring heuristic, which is based on the number
of clusters produced by a pairwise distance threshold and the ratio of the
largest cluster to the second largest across a range of possible thresholds
using sliding windows.

A key advantage of this approach is its autonomy from supplementary data. When a
patient tests positive for HIV, data collection protocols can greatly vary, and
additional data are not always available or consistent. However, by leveraging
only genetic sequence data, AUTO-TUNE eliminates the need for such information.

Consequently, AUTO-TUNE's performance is consistently controlled, irrespective
of the fluctuations seen in data collection protocols after a positive HIV
diagnosis. This level of adaptability demonstrates its suitability for
integration into various contexts related to HIV, and possibly other viral,
cluster detection and response. This versatility underscores the strong
methodological foundation of AUTO-TUNE and its potential utility.

\subsection{When a Score is Below Two}

\section{Conclusion}

\section*{Conflict of Interest Statement}
%All financial, commercial or other relationships that might be perceived by the academic community as representing a potential conflict of interest must be disclosed. If no such relationship exists, authors will be asked to confirm the following statement: 

The authors declare that the research was conducted in the absence of any commercial or financial relationships that could be construed as a potential conflict of interest.

\section*{Author Contributions}

The Author Contributions section is mandatory for all articles, including articles by sole authors. If an appropriate statement is not provided on submission, a standard one will be inserted during the production process. The Author Contributions statement must describe the contributions of individual authors referred to by their initials and, in doing so, all authors agree to be accountable for the content of the work. Please see  \href{https://www.frontiersin.org/guidelines/policies-and-publication-ethics#authorship-and-author-responsibilities}{here} for full authorship criteria.

\section*{Funding}
Details of all funding sources should be provided, including grant numbers if applicable. Please ensure to add all necessary funding information, as after publication this is no longer possible.

\section*{Acknowledgments}
This is a short text to acknowledge the contributions of specific colleagues, institutions, or agencies that aided the efforts of the authors.

\section*{Supplemental Data}
 \href{https://www.frontiersin.org/guidelines/author-guidelines#supplementary-material}{Supplementary Material} should be uploaded separately on submission, if there are Supplementary Figures, please include the caption in the same file as the figure. LaTeX Supplementary Material templates can be found in the Frontiers LaTeX folder.

\section*{Data Availability Statement}

Data are available at GenBank accession numbers
JX160108-JX161480,JX498971-JX498972,JX498976-JX498990,JX498992-JX499018,KU190031-KU190839,KY34691-KY37792,KY883695-KY883762,KY888784-KY888875,KY921717-KY921757,MG434786-MG435347,MG435358-MG436769,MH352627-MH355541,MK25232-MK25548,MN424584-MN427369,MT336755-MT336776,MT368043-MT369927.

\bibliographystyle{Frontiers-Harvard} %  Many Frontiers journals use the Harvard referencing system (Author-date), to find the style and resources for the journal you are submitting to: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal. For Humanities and Social Sciences articles please include page numbers in the in-text citations 
%\bibliographystyle{Frontiers-Vancouver} % Many Frontiers journals use the numbered referencing system, to find the style and resources for the journal you are submitting to: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal
\bibliography{references}
\nocite{*}

%%% Make sure to upload the bib file along with the tex file and PDF
%%% Please see the test.bib file for some examples of references


\section*{Tables}

\begin{table}[h]
\caption{clustuneR Comparison}
\vspace{8pt}
\centering
\label{tab:homophily}
\begin{tabular}{|c|c|c|c|c|c|}
\hline
    Dataset & \multicolumn{2}{c|}{clustuneR} & \multicolumn{3}{c|}{AUTO-TUNE} \\
    \cline{2-6}
     & Threshold & Avg. Homophily & Threshold & Avg. Homophily & Max Score \\
\hline
    Middle Tennessee &  0.0160 & 0.0079 &  0.01431 & 0.0147 & 1.25807 \\
    Seattle & 0.0160 & 0.0259 & 0.01354 & 0.0348 & 1.53325 \\
    Northern Alberta & 0.0104 & -0.0536 & 0.01099 & -0.0448 & 1.01678 \\
\hline
\end{tabular}
\end{table}


\begin{table}[h]
\caption{Threshold Comparison with Prior Publications Citing HIV-TRACE}
\vspace{10pt}
\centering
\label{tab:paperComparison}
\begin{tabular}{|c|c|c|c|c|}
\hline
PMID & Country & Collection Date & Threshold Used & AUTO-TUNE \\
\hline
29975689 & South Africa & 2011-2015 & 2.5\% & 2.584\% \\
30574123 & USA & 1997-2008 & 2\% & 1.848\% \\
32500089 & China & 2008-2015 & 0.5\% & 0.675\% \\
32693608 & Uganda & 2009-2016 & 1.5\% & 1.707\% \\
33281803 & China & 2000-2016 & 0.5\%/0.7\% & 0.676\% \\
33901684 & China & 2008-2012 & 1.5\% & 1.215\% \\
34452506 & Canada & 1996-2017 & 1.5\%/2.5\% & 0.547\% \\
31041344 & USA & 1997-2017 & 1.5\% & 0.927\% \\
\hline
\end{tabular}
\end{table}

\begin{table}[h]
\caption{CRF07\_BC DWH and Panmictic Range at Different Thresholds}
\label{tab:combined}
\centering
\begin{tabular}{|c|c|c|c|c|c|c|}
\hline
& \multicolumn{2}{c|}{Threshold 1.5\%} & \multicolumn{2}{c|}{Threshold 0.76\%} & \multicolumn{2}{c|}{Threshold 0.19\%} \\
\cline{2-7}
Record & DWH & Panmictic Range & DWH & Panmictic Range & DWH & Panmictic Range \\
\hline
MSM & 0.211 & [-0.105, -0.205] & 0.237 & [-0.120,-0.240] & 0.292 & [-0.146, -0.280] \\
Hetero & 0.133 & [-0.092, -0.190] & 0.185 & [-0.100,-0.211] & 0.25 & [-0.093, -0.256] \\
PWID & 0.168 & [-0.001, -0.089] & 0.401 & [-0.005,-0.081] & 0.445 & [-0.012, -0.129] \\
\hline
\end{tabular}
\end{table}

\section{Figure captions}

\begin{figure}[h!]
\caption{ Hola }\label{fig:clustuner}
\end{figure}

\begin{figure}[h!]
\caption{ Hola 2 }\label{fig:paperComparison}
\end{figure}

\begin{figure}[h!]
\caption{ Hola 3 }\label{fig:chinaScorePlot}
\end{figure}


\begin{figure}[h!]
\caption{ Hola 4 }\label{fig:scoreAbove2}
\end{figure}

\end{document}
