%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% This is just an example/guide for you to refer to when submitting manuscripts to Frontiers, it is not mandatory to use Frontiers .cls files nor frontiers.tex  %
% This will only generate the Manuscript, the final article will be typeset by Frontiers after acceptance.   
%                                              %
%                                                                                                                                                         %
% When submitting your files, remember to upload this *tex file, the pdf generated with it, the *bib file (if bibliography is not within the *tex) and all the figures.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% Version 3.4 Generated 2022/06/14 %%%
%%% You will need to have the following packages installed: datetime, fmtcount, etoolbox, fcprefix, which are normally inlcuded in WinEdt. %%%
%%% In http://www.ctan.org/ you can find the packages and how to install them, if necessary. %%%
%%%  NB logo1.jpg is required in the path in order to correctly compile front page header %%%

\documentclass[utf8]{FrontiersinHarvard} % for articles in journals using the Harvard Referencing Style (Author-Date), for Frontiers Reference Styles by Journal: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal
%\documentclass[utf8]{FrontiersinVancouver} % for articles in journals using the Vancouver Reference Style (Numbered), for Frontiers Reference Styles by Journal: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal
%\documentclass[utf8]{frontiersinFPHY_FAMS} % Vancouver Reference Style (Numbered) for articles in the journals "Frontiers in Physics" and "Frontiers in Applied Mathematics and Statistics" 

%\setcitestyle{square} % for articles in the journals "Frontiers in Physics" and "Frontiers in Applied Mathematics and Statistics" 
\usepackage{url,hyperref,lineno,microtype,subcaption}
\usepackage{svg}
\usepackage[onehalfspacing]{setspace}
\usepackage{listings}
\usepackage{rotating}
\usepackage{tabularx}
\usepackage[english]{babel}
\usepackage{booktabs}
\usepackage{longtable}


\newcommand{\TODO}[1]{{\color{red}{#1}}}

\lstset{
  basicstyle=\ttfamily,
  breaklines=true,  % Enable line breaking
  postbreak=\mbox{$\hookrightarrow$}  % Symbol for breakline
}

\lstdefinestyle{BashInputStyle}{
  language=bash,
  basicstyle=\small\sffamily,
  numbers=left,
  numberstyle=\tiny,
  numbersep=5pt,
  frame=single,
  columns=fullflexible,
  backgroundcolor=\color{lightgray},
  linewidth=\linewidth,
  xleftmargin=0.075\linewidth,
  literate={-}{-}1 {'}{{\textquotesingle}}1,
}

\linenumbers


% Leave a blank line between paragraphs instead of using \\


\def\keyFont{\fontsize{8}{11}\helveticabold }
\def\firstAuthorLast{Weaver {et~al.}} %use et al only if is more than 1 author
\def\Authors{Steven Weaver\,$^{1*}$, Vanessa Davila Conn\,$^{2}$, Hannah Verdonk\,$^{1}$, Joel O. Wertheim\,$^{3}$, and Sergei L. Kosakovsky Pond\,$^{1}$}
% Affiliations should be keyed to the author's name with superscript numbers and be listed as follows: Laboratory, Institute, Department, Organization, City, State abbreviation (USA, Canada, Australia), and Country (without detailed address information such as city zip codes or street names).
% If one of the authors has a change of address, list the new address below the correspondence details using a superscript symbol and use the same symbol to indicate the author in the author list.
\def\Address{$^{1}$ Center for Viral Evolution, Temple University, Philadelphia, PA, USA \\
$^{2}$ Center for Research in Infectious Diseases, National Institute of Respiratory Diseases, Mexico City, Mexico  \\ 
$^{3}$ Department of Medicine, University of California San Diego, La Jolla, CA, USA
}
% The Corresponding Author should be marked with an asterisk
% Provide the exact contact address (this time including street name and city zip code) and email of the corresponding author
\def\corrAuthor{Steven Weaver}

\def\corrEmail{sweaver@temple.edu}

\begin{document}
\onecolumn
\firstpage{1}

\title { AUTO-TUNE: FINDING AN OPTIMAL DISTANCE THRESHOLD FOR INFERRING HIV TRANSMISSION CLUSTERS }

\author[\firstAuthorLast ]{\Authors} %This field will be automatically populated
\address{} %This field will be automatically populated
\correspondance{} %This field will be automatically populated

\extraAuth{}% If there are more than 1 corresponding author, comment this line and uncomment the next one.
%\extraAuth{corresponding Author2 \\ Laboratory X2, Institute X2, Department X2, Organization X2, Street X2, City X2 , State XX2 (only USA, Canada and Australia), Zip Code2, X2 Country X2, email2@uni2.edu}

\maketitle

\begin{abstract}

	%%% Leave the Abstract empty if your article does not require one, please see the Summary Table for full details.
	\section{}

	\noindent Molecular surveillance of viral pathogens and inference of
	transmission networks and clusters from genomic data play an increasingly
	important role in public health applications and transmission mitigation
	efforts, especially for HIV-1. For many methods, the genetic distance
	threshold used to connect sequences in the transmission network is a key
	parameter informing the properties of inferred networks. Using a distance
	threshold that is too high can result in a network with many spurious links,
	making it difficult to interpret and analyze. On the other hand, using a distance threshold that
	is too low can result in a network with too few links, which may not capture
	key insights into rapidly growing clusters among patients with shared
	attributes that could benefit from public health intervention measures.
	Published research using the HIV-TRACE software package frequently uses the
	default threshold of $1.5\%$ for HIV pol gene sequences, but in many cases,
	investigators select other threshold parameters (heuristically) to better
	capture the underlying dynamics of the epidemic they are studying.

	Here, we present a heuristic scoring approach for tuning a distance threshold
	adaptively, which seeks to prevent the formation of giant clusters (measured as
	the ratio $R_{12}$ of the sizes of the largest and the second largest cluster),
	and to maximize the number of clusters present in the network.

	We apply our scoring heuristic to outbreaks with different characteristics,
	such as regional or temporal variability, and demonstrate the utility of using
	the scoring mechanism's suggested distance threshold to identify clusters
	exhibiting risk factors that would have otherwise been more difficult to
	identify. For example, while we found that a $1.5\%$ distance threshold is
	typical for US-like epidemics, recent outbreaks like the CRF07\_BC subtype
	among men who have sex with men (MSM) in China have been found to have a lower
	optimal threshold of $0.5\%$ to better capture the transition from injected
	drug use (IDU) to MSM as the primary risk factor. Alternatively, in communities
	surrounding Lake Victoria, where there has been sustained transmission for
	several years, we found that a larger distance threshold is suitable to capture
	a more risk factor-diverse populace with sparse sampling over a longer period
	of time. Such identification may allow for more informed intervention action by
	respective public health officials.

	\tiny
	\keyFont{ \section{Keywords:} molecular epidemiology, HIV, network, transmission cluster, surveillance} %All article types: you may provide up to 8 keywords; at least 5 are mandatory.

\end{abstract}

\section{Introduction}

The use of genomic data to infer and characterize transmission networks of
various pathogens has taken off in the past two decades, with applications to a growing list of microorganisms, including viruses such as HIV \citep{paraskevis_application_2016}, HCV \citep{murphy_molecular_2019} or influenza virus \citep{jombart_reconstructing_2011}, and bacteria such as $M. tuberculosis$ \citep{mai_mycobacterium_2018} or $A. baumanii$ \citep{thoma_challenge_2022}, just to mention some. Notably, genomic surveillance had a prominent role during the COVID-19 pandemic, including the use of sequencing for the study of transmission clusters \citep{von_rotz_systematic_2023,campigotto_utility_2023}.
On the whole, choosing an
appropriate genetic distance threshold is an important part of using a
molecular transmission network to track the spread of rapidly evolving
pathogens \citep{liu_dynamics_2020,rose_persistence_2020}. This distance
threshold defines the degree of genetic closeness required between pathogen
sequences, isolated from two individuals, to suggest them as potential
transmission partners in the network. Using a distance threshold that is too
small can result in a network with many spurious, i.e. epidemiologically
uninformative, links, making it difficult to interpret and analyze. On the
other hand, using a distance threshold that is too large can result in a
network with too few links, underestimating connections between individuals and
making it difficult to accurately track the spread of the disease
\citep{gore_hiv_2022}.

To optimize the utility of inferred transmission networks, it is
important to carefully consider the appropriate distance threshold, $d$. This threshold may
vary depending on the specific disease and the context in which it is spreading.
For example, a highly contagious respiratory illness may require a smaller
$d$ than a less contagious illness that is primarily spread
through direct contact. Viruses are more amenable to molecular studies compared
to bacteria due to their high genetic divergence and compact genomes. Given the
relatively high evolutionary rate of RNA viruses (compared to chromosomal DNA in
humans) detectable genetic fingerprints can be targeted for epidemiological
studies over short time periods\citep{paraskevis_application_2016}.

For chronic infections such as HIV, the most appropriate genetic distance
threshold should be determined according to the characteristics of the epidemic
such as the speed of transmission, and the evolutionary rate of the gene region
analyzed \citep{liu_dynamics_2020}. Sampling density and possible delays between
infection and diagnosis should be considered, since samples close to the time
of seroconversion are more likely to cluster than samples from well after
infection. Lower thresholds will capture the most closely related sequences,
while higher thresholds will capture long-term epidemics and chronically
infected individuals \citep{junqueira_factors_2019}.

Cluster analysis, i.e., identification and analysis of connected network
components, in public health has been used for early identification of
increased transmission \citep{oster_hiv_2021, oster_identifying_2018},
monitoring response to an HIV outbreak \citep{tumpney_human_2020,
	sizemore_using_2020, tookes_rapid_2020}, evaluating the effectiveness of
interventions \citep{peters_hiv_2016,wang_targeting_2015,liu_dynamics_2020} or
predicting clusters that are most likely to grow in the near future
\citep{erly_predictive_2021,ragonnet-cronin_forecasting_2022}. In general, the goal of informative network inference is to strike a balance
between having enough links to accurately track the spread of the disease,
while not having so many links that spurious connections are made and hinder
the public health response. This can be achieved through careful analysis and
consideration of the specific disease and context.

This study introduces AUTO-TUNE, a novel method that offers a systematic
approach to select genetic distance thresholds for molecular HIV transmission
network analysis, based purely on the structure of the collected data. By autonomously optimizing clustering metrics derived from
pairwise genetic distances, AUTO-TUNE has the potential to improve the accuracy
and reliability of network inference, irrespective of data
attributes. The AUTO-TUNE methodology's independence from supplementary data
makes it robust to variations in data collection protocols and enhances its
adaptability to various contexts, including potentially other viral diseases.

\section{Methods}

Assume that there are $S$ aligned genomic sequences (full or partial, e.g. the
HIV-1 pol gene) for a pathogen of interest, each representing the circulating
viral diversity at the time of sampling in a single infected individual. We
shall infer a putative transmission network comprising $S$ nodes, and $E$ links
(edges), where an edge is drawn between a pair of sequences if the genetic
distance between them is below a threshold $d$. In such a network, there will
be $1\leq C \leq S$ connected components (clusters), which are the primary
object of inference. This network inference strategy is used by HIV-TRACE
\citep{kosakovsky_pond_hiv-trace_2018}, where the genetic distance is computed using the
Tamura-Nei (TN93) \citep{tamura_estimation_1993} approach, with a variety of options
controlling how to deal with ambiguous nucleotide bases; for HIV-1 such bases
are informative since they often represent variants co-circulating in the
infected individual at the time of sampling.

We begin by describing an approach to assign a score to each of the choices of
$d$ in a plausible/informative range of distances. Note that while such a range
is continuous, it is sufficient to only consider distance cutoffs that are in
the array of pairwise distances between the sequences, as those are the
cut-points where one or more additional edges will be added to the network as
$d$ is increased.

\subsection{Scoring Heuristic Procedure}

Network threshold selection procedure proceeds as follows:

\begin{enumerate}
	\item{For each candidate threshold $d_L$, in increasing order, ranging from the smallest genetic distance in the dataset, up to either the largest distance or a predetermined maximal threshold, we compute two network statistics: $R_{12}$, the ratio of the size of the largest cluster to the size of the second largest cluster, and $C$ – the number of clusters in the network at this threshold.}

	\item{ A priority score is assigned to each $d_L$. This score measures two properties of the threshold: Does $R_{12}$ jump at $d_L$? How far is the number of clusters $C$ at $d_L$ from the maximal number of clusters over all threshold values? Let there be $N$ overall $d_L$ candidate values, and assume we are examining the ith candidate, $d_L^i$ with $W < i \leq N - W$ ($W$ is a positive integer defined below).

	            \begin{enumerate}
		            \item{The $R_{12}$ jump is computed by looking at the normalized ratio of the mean $R_{12}$ values computed over the leading window $d_L^{i+1}…d_L^{i+W}$ and the trailing window $d_L^{i-W}… d_L^{i-1}$. The width of the window, $W$, is defined as $\min \left( \max \left(\left[\frac{N}{100}\right], 3\right), 30 \right)$. The distribution of ratios is converted to $Z$ scores, and normalized relative to the largest positive $Z$ score across all candidate distances, yielding the jump component of the score.}
		            \item{The number of clusters, $C_i$ at threshold $d_L^i$ is first normalized to $[0,1]$ through $\frac{{C_{max} - C_i}}{{C_{max} -C_{min}}}$ and next gated via a Gompertz function transform ${1-e}^{-e^{-25x+3}}$ This function provides an \emph{ad hoc} means for penalizing having too few clusters relative to the maximum over all ranges. For example, a threshold that yields $95\%$ of the maximal number of clusters receives a score of $0.996$, while a threshold that yields $85\%$ - a score of $0.376$.}
		            \item{The priority score for $d_L^i$ is the sum of the two components defined in (a) and (b).}
	            \end{enumerate}}

	\item{The threshold with the highest priority score will be selected as the suggested automatic distance threshold, if the score is high enough ($1.9$ or more), and either of the two conditions hold.
	            \begin{enumerate}
		            \item{No other thresholds have priority scores of $1.9$ or higher}
		            \item{If other thresholds have priority scores of $1.9$ or higher, then the range of thresholds represented by these options is small (no more than $\log N$ times the mean step between successive $d_L^i$).}
	            \end{enumerate}}

	\item{If no single threshold can be selected in step 3, then the one with the highest priority score is suggested, and an inspection of a plot of scores is recommended to ensure that the threshold is sensible.}
\end{enumerate}

\subsection{Assortativity}

Degree-weighted homophily (DWH) is a measure of similarity between nodes in a
network based on their attributes (such as demographic characteristics or
behaviors) and their degree (i.e., the number of connections they have to other
nodes in the network). It is used to quantify the extent to which nodes with
similar attributes tend to be connected to each other more frequently than
would be expected by chance \citep{golub_how_2012}. DWH is calculated as the ratio
of the observed number of connections between nodes with similar attributes to
the expected number of connections between such nodes, based on their network
degree.

In mathematical terms, it is defined as:
\begin{equation}
	DWH = \frac{W_M + W_C - 2W_X}{\frac{d_{in}}{nodes_{in}^2} + \frac{d_{out}}{nodes_{out}^2}}
\end{equation}

Where
\begin{itemize}
	\item{$W_M$ : Weight of in-group connections}
	\item{$W_C$ : Weight of out-group connections}
	\item{$W_X$ : Weight of cross-group connections}
	\item{$d_{in}$ : In-group degree}
	\item{$d_{out}$ : Out-group degree}
	\item{$nodes_{in}$ : number of in-group nodes}
	\item{$nodes_{out}$ : number of out-group nodes}
\end{itemize}

DWH ranges from $-1$ to $1$. A DWH value of $0$ indicates that there is no more
homophily than expected with chance, while a value of $1$ indicates that there
is perfect homophily (e.g. birds always link to birds, and only birds). A value
of -1 is achieved for perfectly disassortative networks (e.g. a bird never
linking with another bird).

DWH has been used extensively in social network analysis and in the study of
how different attributes are related to the formation of connections between
individuals. It is used as a way to measure the similarity of attributes
between individuals in a network. Additionally, randomization is performed by
shuffling attribute labels among nodes, then performing DWH computation. This
is useful in creating a null distribution of DWH scores under random mixing. A
panmictic range is reported by shuffling attributes multiple times and
reporting the minimum and maximum score.

\subsection{Implementation}

The software implementation involves a step-by-step process that utilizes the
HIV-TRACE suite of packages. It starts with calculating pairwise distances wth
the {\tt tn93} tool and a supplied multiple sequence alignment. Thus generated
pairwise distances are supplied to the {\tt hivnetworkcsv} script while
providing the {\tt -A} keyword argument. A brief outline of the software's
implementation is as follows

\begin{enumerate}

	\item{ Calculate pairwise distances: The user first calculates the pairwise distances using the {\tt tn93} fast pairwise distance calculator, providing the maximum threshold value to consider ($0.03$ in this case) and the input FASTA file. The command for this step is
	            \begin{lstlisting}[style=BashInputStyle]
 tn93 -t 0.030 pol.fasta > pairwise_distances.15.tn93.csv
\end{lstlisting}

	            Please note that the threshold should include the maximal range one is
	            intending to test.

	      }

	\item {Compute distance threshold scores: The {\tt hivnetworkcsv} script is then executed with the required input file, format, and autotune option to generate a tab-separated output file, as shown below
	      \begin{lstlisting}[style=BashInputStyle]
 hivnetworkcsv -i pairwise_distances.15.tn93.csv -f plain -A 0 > autotune_report.tsv
\end{lstlisting}
	      }

	\item {Visualize the report: Users can upload the generated {\tt autotune\_report.tsv} file to \\
	      \url{http://autotune.datamonkey.org/analyze} for visualization and further analysis of the data. This web-based site extends the Datamonkey platform \citep{weaver_datamonkey_2018} to provide an interactive environment to explore scores and other metrics across the range of tested outputs. }

	\item {Run HIV-TRACE: Once AUTO-TUNEd threshold(s) are settled upon after review, the user runs the HIV-TRACE command with the appropriate input FASTA file, distance threshold, and other required arguments. The output is saved as a JSON file. An example command is
	      \begin{lstlisting}[style=BashInputStyle]
hivtrace -i ./INPUT.FASTA -a resolve -r HXB2_prrt -t < autotune_threshold > -m 500 -g .05 > hivtrace.results.json
	\end{lstlisting}
	      }

	      \subsubsection{Optional : Compute Assortativity Metrics}

	      \item{ Annotate results: The {\tt hivnetworkannotate} script is used to annotate the results obtained from the HIV-TRACE step with attributes. The script takes the JSON results file, node attributes file, schema file, and a resolve flag as input.
	                  \begin{lstlisting}[style=BashInputStyle]
hivnetworkannotate -n hivtrace.results.json -a node_attributes.json -g schema.json -r
	\end{lstlisting}
	                  For more information, users can refer to the {\tt hivnetworkannotate}
	                  documentation. }

	      \item{ Analyze the results with DWH: After the results file has been annotated, the user can proceed to the assortativity page, \url{http://autotune.datamonkey.org/assortativity}, for further analysis of the output. }

	      AUTO-TUNE is readily accessible on GitHub as part of the {\tt hivclustering}
	      repository (\url{https://github.com/veg/hivclustering})). It is integrated into
	      the command-line interface of the software as the {\tt -A} or {\tt
			      --auto-profile} argument. {\tt hivclustering} is a key component of the
	      HIV-TRACE suite of tools, a resource for the inference, analysis, and
	      visualization of HIV transmission networks.

	      The Degree Weighted Homophily (DWH) calculation tool, an integral component of
	      the assortativity step, is developed using {\tt TypeScript}, a statically typed
	      superset of {\tt JavaScript} that ensures robustness and scalability. In an
	      effort to promote accessibility and ease of integration, the DWH tool is
	      packaged and distributed through the Node Package Manager (npm), enabling
	      researchers and developers to conveniently incorporate this advanced analytical
	      tool into their own projects and workflows. DWH can be used in-browser or as a
	      command-line tool, allowing researchers and developers to employ the tool in an
	      interactive command-line interface or integrate it into larger software
	      applications, thus catering to a diverse array of technical needs and
	      preferences. Instructions for usage and installation are found on Github
	      (\url{https://github.com/veg/dwh}).

\end{enumerate}

The described workflow offers a systematic approach to analyze potential
distance thresholds for one's data with AUTO-TUNE, from calculating pairwise
distances to visualizing and annotating results.

\subsection{Visualization}

Visualizations of AUTO-TUNE results are accessible at
\url{http://autotune.datamonkey.org/analyze}. It is a dynamic and interactive
web-based platform that offers visualization and analysis of results generated
by AUTO-TUNE. The website provides a comprehensive view of the data by
generating various plots across candidate distance thresholds. These include a
score plot, allowing users to identify trends and anomalies across the full
range of thresholds. Additionally, it generates a graph showing the number of
clusters across candidate thresholds, one of the components that contribute to
the score. The site also includes an R1/R2 plot that displays the ratio of the
largest cluster to the second largest cluster across candidate thresholds,
which is the other metric that contributes to the scoring heuristic. \TODO{this
	could benefit from referencing a figure showing such plots}

An assortativity tool is available at
\url{http://autotune.datamonkey.org/assortativity}, and is an advanced
analytical tool engineered to facilitate the calculation of Degree Weighted
Homophily (DWH) values. It utilizes the DWH NPM package to generate a tabular
representation of DWH values corresponding to each value for a selected
attribute annotation, providing an exhaustive examination of the
interrelationships for the field. A notable feature is the computation of the
panmictic range, which involves a label permutation test to generate the null
distribution of DWH values. This feature establishes a comparative baseline
that aids in determining the significance of homophily versus what would be
expected by chance. Lastly, the site also provides a plot of the fraction of
pairwise connections, normalized by degree, for each value pertinent to the
selected field. This visual depiction facilitates an intuitive comprehension of
the distribution and interconnections within the dataset.

The site aims to offer a user-friendly interface for data visualization,
playing an important role in interpreting and understanding AUTO-TUNE's output
data. The visualization code is available on Github
(\url{https://github.com/stevenweaver/autotune-app/}).

\subsection{Comparisons with previously published analyses}

In conducting our comparisons with the established {\tt clustuneR} method
\citep{chato_public_2020}, we procured datasets from \citep{wolf_short_2017} and
\citep{vrancken_multi-faceted_2017} utilizing the identical approach delineated
in \cite{chato_public_2020}. These datasets, namely Middle Tennessee, Seattle,
and Alberta, were processed using the workflow prescribed in Section 2.3. This
enabled us to determine an optimal threshold for each dataset using AUTO-TUNE.
We further executed the command as detailed in step 4 of Section 2.3, deploying
thresholds previously established as optimal by \citep{chato_public_2020}.

To perform comparisons, we computed the average degree-weighted homophily score
over a set of three-year sliding windows. Specifically, the homophily among
nodes was calculated for a collection of date ranges as follows:

\begin{equation} \bar{H} = \frac{1}{N}\sum_{i=1}^{N} H(w_i) \end{equation}

where $\bar{H}$ represents the average degree-weighted homophily score, $N$ is
the total number of sliding windows, $H(w_i)$ is the homophily score for the
$i$-th window, and the windows $w_i$ correspond to the date ranges, e.g.,
'2012-2015', '2013-2016', '2014-2017', etc. This methodology allowed us to
compare the "best thresholds" derived from our proposed AUTO-TUNE method
against those defined as optimal in \citep{chato_public_2020}.

Second, we set out to compare the thresholds obtained in numerous published
investigations with those obtained by AUTO-TUNE. To select the data sets for
this analysis, we conducted a scientific literature search to identify studies
focused on HIV networks for public health purposes. We then filtered the
studies that utilized HIV-TRACE to infer genetic networks and had publicly
available sequences. Thus, we attempted to include studies from different
countries and regions, enabling us to assess the performance of our method
across various epidemic contexts, risk groups, and network sizes in real-data
sets that used variable clustering thresholds.

In order to evaluate the influence of sampling density on the genetic distance
threshold as determined by AUTO-TUNE, we implemented a strategy of random
subsampling from the original dataset sourced from \citep{rhee_national_2019}.
This study was selected due to its satisfactory AUTO-TUNE score when utilized
in its entirety, as well as its inherent design as a Geographically-Stratified
set of 716 Pol Subtype/CRF (GSPS) reference sequence dataset. The dataset,
which comprises 6034 samples gathered between 1959 and 2016, was subjected to
random subsampling ten times at proportions of $25\%$, $50\%$, and $75\%$ of
the original sample size. For each subsample, the optimal threshold and
associated scores were determined via AUTO-TUNE.

\section{Results}

\subsection{Evaluating Performance using DWH}

Non-random mixing and attribute-based homophily are intrinsic characteristics
of human social networks and can be expected across transmission network,
particularly in the context of HIV transmission. People frequently engage in
relationships with those who share similar attributes or behaviors, such as
risk factors (e.g., PWID, MSM). The effect of these nonrandom mixing patterns
on the genetic diversity of HIV-1 has not only been extensively explored via
modeling and simulations \citep{goodreau_assessing_2006}, but the structure of
sexual contact networks has been found to directly influence pathogen
phylogenies \citep{robinson_how_2013}. Additionally, phylogenetic analysis of
HIV type 1 sequences has revealed distinct grouping patterns based on risk
behaviors \citep{holmes_molecular_1995}. Therefore, a higher homophily can be
considered a more accurate and realistic representation of these phenomena
within the transmission network.

To assess the performance of an AUTO-TUNEd optimized threshold using
degree-weighted homophily, we first evaluated a CRF07\_BC network with data
from China. We used $8178$ HIV-1 CRF07\_BC pol sequences collected by a
national survey in China to construct longitudinal transmission networks. Each
pol sequence was annotated with a risk factor detailing whether the patient was
heterosexual (Hetero), Person With Injected Drug Use (PWID), or Men who have
Sex with Men (MSM), among other attributes. 

When we analyze the dataset with AUTO-TUNE, similar high scores were achieved
with 0.76\% and 0.19\% thresholds, at scores 1.137 and 1.030, respectively.
Notably, the DWH scores for PWID exhibited a significant surge at these
thresholds, indicating a robust pattern of increased PWID homophily. The close
proximity of AUTO-TUNE scores and the consistent rise in PWID homophily at
0.76\% and 0.19\% thresholds suggest comparable performance at these thresholds
compared to the default 1.5\% threshold, suggesting that these thresholds might
be more effective in representing homophilic relationships in this network. At
each threshold—1.5\%, 0.76\%, and 0.19\%—all DWH scores for the risk groups
(MSM, Hetero, and PWID) lie outside their respective panmictic ranges. This
consistently indicates non-random mixing and attribute-based homophily across
the network. Detailed results are in Table~\ref{table:combined}. 

We next evaluated \citep{rhee_national_2019} with DWH \ref{table:rhee}. When
assessing assortative mixing patterns across countries, China exhibited the
most significant change in DWH values from the default 1.5\% threshold to the
AUTO-TUNEd 1.699\% threshold, shifting from -0.166 to 0.116. The shift in
China's DWH placed its value outside its panmictic range. The implication being
that, with an AUTO-TUNE threshold, assortative mixing patterns are captures
that would otherwise be missed. Thailand also exhibited the same pattern in
this dataset. Notably, no countries registered a decrease in their DWH value,
despite the increased threshold.
\label{fig:scoreAbove2}



\subsection{Comparison with clustuneR}

We compared results to {\tt clustuneR}, which employs the recency of sample
collection or diagnosis as individual-level weights in a predictive model to
estimate the growth of HIV clusters. The thresholds deemed optimal by {\tt
		clustuneR} were found by finding the minimum GAIC (generalized AIC) across
candidate distances between $0$ and $0.04$ in steps of $8 \times 10^{-4}$. GAIC
is the difference between a null model that is only influenced by cluster size,
and a weighted model that includes individual-level attributes among known
cases in the cluster. Using the minimum GAIC metric, it was found that $0.016$
was the optimal threshold for Tennessee and Seattle, and $0.0104$ for Northern
Alberta.

In contrast, AUTO-TUNE does not incorporate any attribute data in its scoring
heuristic. Instead, it relies on clustering metrics constructed purely from
pairwise distances between sequences. Using the same datasets analyzed by {\tt
		clustuneR} \citep{chato_public_2020}, AUTO-TUNE found the thresholds with the
highest scores to be $0.01431$ for Middle Tennessee, $0.01354$ for Seattle, and
$0.01099$ for Northern Alberta. \autoref{tab:homophily}. 

\TODO{discuss how our thresholds perform vs the published thresholds; look at the GAIC plots}
\TODO{Add performance metrics}

Performance of the inferred optimal thresholds were determined using an average
degree-weighted homophily (DWH) score across 3-year collection date windows
starting from the oldest collection year for each respective dataset, as that
is the metadata that was consistently available across all three datasets and
was the attribute of focus used by {\tt clustuneR}. DWH in this case measures
the affinity for nodes within the network to link with other nodes in the same
collection date window. For example, samples collected between 2012-2015
linking more often with other samples within the same time window would result
in a higher DWH score. It was found that, across all three datasets, using the
threshold with the highest score reported by AUTO-TUNE resulted in a higher
average DWH score across all three datasets when compared to the clustuneR
derived threshold. 

% Add something about Wilcoxon test
% The p-values for the Wilcoxon sign test are 0.0018 for Seattle, 0.999 for Tennessee, and 0.1797 for Canada.



When reviewing scores across all candidate thresholds with AUTO-TUNE
\label{fig:clustuner}, none of the three datasets reached one confident score
over any other. The most confident score was achieved by the Seattle dataset at
$1.53$, then Tennessee with a high score of $1.26$, and lastly Canada at
$1.02$. When finding peaks, a second peak in Seattle denotes that $0.01166$ may
also be an optimal threshold to consider, as its score, $1.52976$, is only
$0.003$ less than the highest score. The optimal score for Tennessee and Canada
are a bit more dubious, as there are multiple peaks within close scores of each
other. Indeed, after applying a $0.75$ minimum score threshold after visual
inspection for peak ranges, standard deviation among scores were $0.0475$,
$0.0089$, and $0.0084$ for Seattle, northern Alberta, and Tennessee,
respectively. This implies there may be multiple thresholds that would be
considered reasonable, and downstream homophily metrics with attributes may aid
in coming to a decision.

\subsection{Comparison with Prior Publications Citing HIV-TRACE}

% It is problematic to tune this threshold and most analyses use an ad-hoc threshold or evaluate sensitivity over a range of thresholds.

Next, we curated publications citing HIV-TRACE that also had publicly
accessible data associated with the study \citep{rhee_national_2019,
	brenner_role_2021, h_acquisition_2021, liu_dynamics_2020, bbosa_short_2020,
	yan_central_2020, dalai_combining_2018, sivay_hiv-1_2018}. We found that a
variety of different ways has been used to determine distance thresholds: using
the precedent set by the CDC for detecting recent and rapid clusters
\citep{yan_central_2020}; using thresholds from other studies
\citep{sivay_hiv-1_2018}; visual inspection of the number of clusters and nodes
across candidate distance thresholds \citep{liu_dynamics_2020}. Because these
thresholds were determined largely qualitatively, they tended to be round
numbers and only somewhat tuned to their respective geographic region of
research. \autoref{tab:paperComparison} When AUTO-TUNE was applied to these
datasets, a range of dataset-specific thresholds was inferred. This analysis
demonstrated that network inference can be highly sensitive to relatively small
changes in the distance threshold. For example, when reviewing the distribution
of scores for the dataset used by \citep{dalai_combining_2018}, the score is
exactly $2$ at $0.01848$, but at thresholds tested at just $0.00002$
difference, scores drop precipitously to $1.638$ and $0.826$ for candidates
$0.01846$ and $0.0185$, respectively. Another example of a seemingly close
threshold yet perhaps not optimal is found with \citep{bbosa_short_2020}. While
no score across candidate thresholds was found to be above $1.9$, a high score
was found at distance $1.707$, with $1.2415$. Contrast this with the threshold
used, $1.5\%$, with a score of $0.0124$. \TODO{I would add some specifics here:
	what is $R_12$ and $C$ for those thresholds, for example? The composite score
	is difficult to appreciate intutitvely}


\subsection{The Effect of Subsampling on Optimal Thresholds and AUTO-TUNE Scores}

\TODO{Also should include some motivation}

Next, we evaluated the performance of AUTO-TUNE when subsampling a dataset.
Since the \citep{rhee_national_2019} dataset exhibited a clear optimal peak, we
used the dataset for analysis, and randomly sampled $10$ times from the entire
dataset at $25\%$, $50\%$, and $75\%$ each. The original full dataset
confidently determined $0.01699$ (AUTO-TUNE score $1.9998$).

Sampling at $25\%$ yielded a mean top threshold of $0.021509$, median at
$0.019765$, and standard deviation of $0.004388$ \label{fig:subsampling}.
$50\%$ yielded $0.018581$ and $0.01871$ mean and median, respectively with a
standard deviation of $0.001629$. Finally, $75\%$ calculated mean is
approximately $0.017403$, with a median of approximately $0.01699$. The
standard deviation was $0.000924$.

\TODO{explain why the trend of HIGHER thresholds for the SPARSER sampling makes sense}

As the proportion increased from $25\%$ to $50\%$ and $75\%$, observable shifts
were also noted in the mean, median, and standard deviation of the AUTO-TUNE
scores. At $25\%$, the mean and median scores were $1.5585$ and $1.5014$
respectively, with a standard deviation of $0.3568$. At $50\%$, both mean and
median scores significantly increased to $1.8171$ and $1.9191$ respectively,
and the standard deviation dropped to $0.2482$. Upon reaching an AUTO-TUNE of
$75\%$, the mean and median scores rose further to $1.9870$ and $1.9997$
respectively, while the standard deviation shrank substantially to $0.0364$,
indicating higher consistency in scores.

As the sample proportion increased, an upward trend was noted in average
AUTO-TUNE scores. Additionally, the standard deviation reduced significantly
with sample proportion. This implies that as sampling becomes denser, AUTO-TUNE
will become more confident in determining the optimal threshold for a
particular dataset.

\TODO{I still think that being able to show what fraction of the nodes connected in the full network are connected in the subsampled networks;
	if you have network JSONs, I can check that pretty quickly}

%TODO: Add supplementary figure of all score plots

\section{Discussion}

AUTO-TUNE addresses the challenge of selecting setting an appropriate genetic
distance threshold in HIV transmission network analysis by implementing a
heuristic scoring system that is based on two desirable properties of networks
generated by candidate genetic distance thresholds: a large number of clusters,
and the absence of a giant component. Too few small clusters are a hallmark of
an excessively low threshold, and a giant cluster that includes too many
sequences is a hallmark of an excessively high threshold. In the application of
AUTO-TUNE to various datasets, the results demonstrated its efficacy across
different epidemic contexts, risk groups, and network sizes. \TODO{Not sure
	what efficacy means} AUTO-TUNE consistently selected thresholds that were
comparable \TODO{comparable how?}, if not better \TODO{better how?}, to those
manually chosen in prior studies using the same data, illustrating the value of
a more systematic, automated, and data-adaptive approach.

For example, the results of our study suggest that AUTO-TUNE, which relies
solely on clustering metrics from pairwise distances, could be an effective
alternative to other distance-based methods, such as {\tt clustuneR} while less
time-consuming and possessing a gentle learning curve, which makes it easy to
use by personnel not specialized in bioinformatics and computer science.
Furthermore, the simplicity of the method represents an advantage over
phylogenetic methods where, in addition to the calculation of genetic
distances, it must also determine a support/distance threshold where a
rationale for the selection of these thresholds is rarely provided
\citep{junqueira_factors_2019}.

AUTO-TUNE generated thresholds for all three examined datasets (Middle
Tennessee, Seattle, and Northern Alberta) that outperformed {\tt clustuneR}
using DWH on 3-year collection date windows across all three datasets. This
indicates that even without incorporating attribute data, AUTO-TUNE's scoring
heuristic could provide reliable thresholds for HIV clusters. However, for the
determination of the optimal genetic distance threshold, time-related and
context-specific factors might need to be considered if there is no significant
score for any one candidate threshold, especially if there are multiple peaks.
For example, during HIV outbreaks in injection drug users (that usually occur
over several months), it may be more appropriate to use the shorter genetic
distance threshold \citep{peters_hiv_2016,campbell_detailed_2017} between
multiple high-scoring thresholds. On the contrary, larger and more extended
epidemics over time exhibit a tendency toward larger genetic distance
thresholds in order to capture transmission than younger epidemics and less
densely sampled epidemic investigations
\citep{patil_exploring_2022,leung_molecular_2019,di_giallonardo_subtype-specific_2021}.

Our evaluation of publications citing HIV-TRACE revealed the largely
qualitative determination of distance thresholds. This approach may result in
less accurate or suboptimal thresholds due to a lack of systematic analysis. In
contrast, AUTO-TUNE offers a more systematic and granular approach to threshold
selection, with our findings demonstrating that even minor adjustments to the
distance can drastically change the score. Therefore, using AUTO-TUNE could
potentially improve the quality of HIV clustering and transmission network
studies.

The Degree-Weighted Homophily (DWH) evaluation showed that AUTO-TUNE could
improve network quality based on specific attributes, such as risk factor,
which is an important part of HIV studies and informing prevention measures
\citep{potterat_risk_2002,fujimoto_methodological_2021}. For example, the use of
AUTO-TUNE resulted in an increased DWH among the MSM, Hetero, and PWID groups
when analyzing a CRF07\_BC network. Additionally, the results from the Rhee et
al. dataset also demonstrated AUTO-TUNE's ability to improve DWH
geographically, enhancing the network's ability to accurately reflect
transmission dynamics.

Our analysis of AUTO-TUNE's performance on subsamples of a dataset revealed its
sensitivity to sample size. The results indicated a correlation between
increased sample size and higher average AUTO-TUNE scores, as well as lower
score variability. This suggests that denser sampling could enhance AUTO-TUNE's
ability to determine the optimal threshold for a dataset. Further studies might
be needed to establish the minimum sample size required for reliable threshold
determination.

\subsection{When a Score is Below 1.9}

\TODO{add some text to explain that multiple scores at different thresholds could be indicative of inherently different scales in the network, e.g. global AND local combined into one}

The use of AUTO-TUNE, while offering a method for automated threshold
selection, may not always provide a single, decisive score that unambiguously
determines the optimal threshold. In certain situations, several candidate
thresholds may yield similar AUTO-TUNE scores, making it difficult to single
out one as the clear-cut 'optimal' threshold. In these scenarios, the process
of threshold selection becomes more nuanced and requires a deeper analysis. The
plot of AUTO-TUNE scores across candidate thresholds can serve as a valuable
tool in these cases. For instance, researchers could identify a range of
thresholds that all produce similar scores, suggesting that the specific choice
of threshold within this range may not significantly impact the resulting
network. Moreover, combining AUTO-TUNE with the DWH measure can enhance the
interpretation of such plots. By considering how assortativity changes across
the range of candidates, researchers can make more informed decisions about the
appropriate choice. If there is a certain threshold at which the DWH measure
noticeably changes for an attribute of interest, this could suggest a
meaningful shift in the network structure that would be worth considering when
selecting a threshold. The symbiotic approach of combining AUTO-TUNE scores,
DWH measure, and visual analysis of score plots provides a more nuanced method
for threshold selection when no clear optimal threshold emerges from the
AUTO-TUNE scores alone.

The AUTO-TUNE methodology has several limitations. First, even though it
provides the advantage of operating without the need for metadata, the size and
the subgenomic region analyzed may affect the accuracy of transmission
inference \citep{junqueira_factors_2019}. Second, our analysis of AUTO-TUNE's
performance on subsamples of a dataset revealed its sensitivity to sample size,
as the performance of the method can be affected by sampling density, improving
the reliability of the test as the sampling density increases (figure X).
However, our results were consistent with previous studies, which have
suggested an optimal sampling density of $50-70\%$ for HIV-1 cluster
analysis\citep{novitsky_impact_2014}. Third, even when it provides an insight of
the optimal threshold to analyze a network, the supplied information might
still need validation by experts, especially when no clear threshold is
identified. In this case, it has been recommended to combine genetic data with
clinical and sociodemographic information for a better characterization of the
network structure. Finally, the performance of the method needs to be assessed
in pathogens different from HIV, leading to opportunities for future research.

\section{Conclusion}

AUTO-TUNE operates solely utilizing genetic sequence data to ascertain a
decisive threshold. It employs a scoring heuristic, which is based on the
number of clusters produced by a pairwise distance threshold and the ratio of
the largest cluster to the second largest across a range of possible thresholds
using sliding windows.

A key advantage of this approach is its autonomy from supplementary data. When
a patient tests positive for HIV, data collection protocols can greatly vary,
and additional data are not always available or consistent. However, by
leveraging only genetic sequence data, AUTO-TUNE eliminates the need for such
information.

Consequently, AUTO-TUNE's performance is consistently controlled, irrespective
of the fluctuations seen in data collection protocols after a positive HIV
diagnosis. This level of adaptability demonstrates its suitability for
integration into various contexts related to HIV, and possibly other viral
cluster detection and response protocols. This versatility underscores the
strong methodological foundation of AUTO-TUNE and its potential utility.

\section*{Conflict of Interest Statement}
%All financial, commercial or other relationships that might be perceived by the academic community as representing a potential conflict of interest must be disclosed. If no such relationship exists, authors will be asked to confirm the following statement: 

The authors declare that the research was conducted in the absence of any
commercial or financial relationships that could be construed as a potential
conflict of interest.

\section*{Author Contributions}

The Author Contributions section is mandatory for all articles, including
articles by sole authors. If an appropriate statement is not provided on
submission, a standard one will be inserted during the production process. The
Author Contributions statement must describe the contributions of individual
authors referred to by their initials and, in doing so, all authors agree to be
accountable for the content of the work. Please see
\href{https://www.frontiersin.org/guidelines/policies-and-publication-ethics#authorship-and-author-responsibilities}{here}
for full authorship criteria.

\section*{Funding}
Details of all funding sources should be provided, including grant numbers if applicable. Please ensure to add all necessary funding information, as after publication this is no longer possible.

\section*{Acknowledgments}
This is a short text to acknowledge the contributions of specific colleagues, institutions, or agencies that aided the efforts of the authors.

\section*{Supplemental Data}
\href{https://www.frontiersin.org/guidelines/author-guidelines#supplementary-material}{Supplementary Material} should be uploaded separately on submission, if there are Supplementary Figures, please include the caption in the same file as the figure. LaTeX Supplementary Material templates can be found in the Frontiers LaTeX folder.

\section*{Data Availability Statement}

Data are available at GenBank accession numbers
JX160108-JX161480,JX498971-JX498972,JX498976-JX498990,JX498992-JX499018,KU190031-KU190839,KY34691-KY37792,KY883695-KY883762,KY888784-KY888875,KY921717-KY921757,MG434786-MG435347,MG435358-MG436769,MH352627-MH355541,MK25232-MK25548,MN424584-MN427369,MT336755-MT336776,MT368043-MT369927.

\bibliographystyle{Frontiers-Harvard} %  Many Frontiers journals use the Harvard referencing system (Author-date), to find the style and resources for the journal you are submitting to: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal. For Humanities and Social Sciences articles please include page numbers in the in-text citations 
%\bibliographystyle{Frontiers-Vancouver} % Many Frontiers journals use the numbered referencing system, to find the style and resources for the journal you are submitting to: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal
\bibliography{references}
\nocite{*}

%%% Make sure to upload the bib file along with the tex file and PDF
%%% Please see the test.bib file for some examples of references

\section*{Tables}

\TODO{Add p-values for DWH, hi Steve}

\begin{table}[h]
	\caption{clustuneR Comparison}
	\vspace{8pt}
	\centering
	\label{tab:homophily}
	\begin{tabular}{|c|c|c|c|c|c|}
		\hline
		Dataset          & \multicolumn{2}{c|}{clustuneR} & \multicolumn{3}{c|}{AUTO-TUNE}                                          \\
		\cline{2-6}
		                 & Threshold                      & Avg. Homophily                 & Threshold & Avg. Homophily & Max Score \\
		\hline
		Middle Tennessee & 0.0160                         & 0.0079                         & 0.01431   & 0.0147         & 1.25807   \\
		Seattle          & 0.0160                         & 0.0259                         & 0.01354   & 0.0348         & 1.53325   \\
		Northern Alberta & 0.0104                         & -0.0536                        & 0.01099   & -0.0448        & 1.01678   \\
		\hline
	\end{tabular}
\end{table}

\begin{table}[h]
	\caption{Threshold Comparison with Prior Publications Citing HIV-TRACE}
	\vspace{10pt}
	\centering
	\label{tab:paperComparison}
	\begin{tabular}{|c|c|c|c|c|}
		\hline
		PMID     & Country      & Collection Date & Threshold Used & AUTO-TUNE \\
		\hline
		29975689 & South Africa & 2011-2015       & 2.5\%          & 2.584\%   \\
		30574123 & USA          & 1997-2008       & 2\%            & 1.848\%   \\
		32500089 & China        & 2008-2015       & 0.5\%          & 0.675\%   \\
		32693608 & Uganda       & 2009-2016       & 1.5\%          & 1.707\%   \\
		33281803 & China        & 2000-2016       & 0.5\%/0.7\%    & 0.676\%   \\
		33901684 & China        & 2008-2012       & 1.5\%          & 1.215\%   \\
		34452506 & Canada       & 1996-2017       & 1.5\%/2.5\%    & 0.547\%   \\
		31041344 & USA          & 1997-2017       & 1.5\%          & 0.927\%   \\
		\hline
	\end{tabular}
\end{table}


\begin{table}[h!]
\centering
\begin{tabularx}{\textwidth}{|X|X|X|X|X|X|}
\hline
Threshold & AUTO-TUNE Score & Nodes & PWID & MSM & Hetero \\
\hline
1.5\% & 0.029 & 5923 & 559 & 3371 & 1993 \\
0.76\% & 1.1369 & 3537 & 236 & 2271 & 1030 \\
0.19\% & 1.0303 & 1654 & 151 & 1075 & 428 \\
\hline
\end{tabularx}
\caption{CRF07\_BC Nodes Count at Different Thresholds}
\label{table:combined}
\end{table}

\begin{table}[h!]
\centering
\begin{tabularx}{\textwidth}{|X|X|X|X|X|}
\hline
Threshold & Risk Group & DWH & Panmictic Min & Panmictic Max \\
\hline
1.5\% & MSM & 0.211 & -0.085 & -0.213 \\
1.5\% & Hetero & 0.133 & -0.087 & -0.190 \\
1.5\% & PWID & 0.168 & 0.002 & -0.091 \\
\hline
0.76\% & MSM & 0.237 & -0.120 & -0.240 \\
0.76\% & Hetero & 0.185 & -0.100 & -0.211 \\
0.76\% & PWID & 0.401 & -0.005 & -0.081 \\
\hline
0.19\% & MSM & 0.292 & -0.146 & -0.280 \\
0.19\% & Hetero & 0.250 & -0.093 & -0.256 \\
0.19\% & PWID & 0.445 & -0.012 & -0.129 \\
\hline
\end{tabularx}
\caption{Panmictic Ranges for CRF07\_BC DWH at Different Thresholds}
\label{table:panmictic}
\end{table}

\begin{table}[h!]
\centering
\begin{tabularx}{\textwidth}{|X|X|X|X|X|X|}
\hline
Threshold & AUTO-TUNE Score & Nodes Captured & Total Nodes \\
\hline
1.5\% & 0.9782 & 1351 & 6034 \\
1.699\% & 1.9998 & 1592 & 6034 \\
\hline
\end{tabularx}
\caption{AUTO-TUNE values with data from \cite{rhee_national_2019}}.
\label{table:rhee}
\end{table}


\begin{table}[h!]
\centering
\begin{tabularx}{\textwidth}{|l|X|X|X|X|X|X|X|}
\hline
\textbf{Country} & \textbf{DWH (1.5\%)} & \textbf{Panmictic Min (1.5\%)} & \textbf{Panmictic Max (1.5\%)} & \textbf{DWH (1.699\%)} & \textbf{Panmictic Min (1.699\%)} & \textbf{Panmictic Max (1.699\%)} & \textbf{DWH Difference} \\
\hline
China & -0.166 & 0.015 & -0.222 & 0.116 & 0.004 & -0.256 & 0.282 \\
Thailand & -0.051 & -0.015 & -0.129 & 0.132 & 0.018 & -0.157 & 0.183 \\
Ukraine & -0.129 & -0.015 & -0.129 & -0.033 & 0.001 & -0.095 & 0.096 \\
Uzbekistan & -0.129 & -0.015 & -0.129 & -0.033 & 0.001 & -0.095 & 0.096 \\
Spain & -0.097 & -0.015 & -0.129 & -0.011 & 0.008 & -0.202 & 0.086\\
Australia & -0.097 & -0.015 & -0.129 & -0.030 & 0.001 & -0.095 & 0.067\\
Angola & -0.097 & -0.015 & -0.129 & -0.045 & 0.001 & -0.095 & 0.052\\
India & -0.097 & -0.015 & -0.129 & -0.045 & 0.001 & -0.095 & 0.052\\
Belarus & -0.097 & -0.015 & -0.129 & -0.045 & 0.001 & -0.095 & 0.052\\
Japan & -0.031 & -0.015 & -0.129 & -0.007 & 0.001 & -0.095 & 0.024\\
Indonesia & -0.015 & -0.015 & -0.129 & -0.007 & 0.001 & -0.095 & 0.008\\
US & -0.051 & -0.015 & -0.129 & -0.043 & 0.001 & -0.095 & 0.008\\
Viet Nam & -0.015 & -0.015 & -0.129 & -0.007 & 0.001 & -0.095 & 0.008\\
Russia & -0.141 & 0.015 & -0.222 & -0.134 & 0.027 & -0.157 & 0.007\\
Kazakhstan & -0.038 & -0.015 & -0.129 & -0.033 & 0.001 & -0.095 & 0.005\\
France & -0.097 & -0.015 & -0.129 & -0.095 & 0.001 & -0.095 & 0.002\\
Cyprus & N/A & N/A & N/A & -0.011 & 0.001 & -0.095 & N/A\\
Georgia & N/A & N/A & N/A & -0.011 & 0.001 & -0.095 & N/A\\
South Africa & N/A & N/A & N/A & -0.030 & 0.001 & -0.095 & N/A\\
Canada & N/A & N/A & N/A & -0.003 & 0.001 & -0.095 & N/A\\
Taiwan & N/A & N/A & N/A & -0.045 & 0.001 & -0.095 & N/A\\
Senegal & N/A & N/A & N/A & 0.499 & 0.043 & -0.137 & N/A\\
Estonia & N/A & N/A & N/A & -0.045 & 0.001 & -0.095 & N/A\\
Singapore & N/A & N/A & N/A & -0.045 & 0.001 & -0.095 & N/A \\
Malaysia & N/A & N/A & N/A & -0.045 & 0.001 & -0.095 & N/A \\
Germany & N/A & N/A & N/A & -0.049 & 0.043 & -0.117 & N/A \\
\hline
\end{tabularx}
  \caption{Comparison of DWH values for different countries with 1.5\% and AUTO-TUNE inferred threshold of 1.699\% for \cite{rhee_national_2019}.}
\label{tab:DWH_comparison}
\end{table}


\begin{longtable}{llcc}
\toprule
Location & Record & DWH\_autotune & DWH\_clustuner \\
\midrule
\endfirsthead
\toprule
Location & Record & DWH\_autotune & DWH\_clustuner \\
\midrule
\endhead
Seattle & 1998-2000 & 0.089 & 0.08 \\
% Continue with other Seattle records
Seattle & AUTO-TUNE AVG & 0.0348125 & clustuneR AVG & 0.025875 \\
Tennessee & 1999-2001 & -0.013 & -0.012 \\
% Continue with other Tennessee records
Tennessee & AUTO-TUNE AVG & 0.014764706 & clustuneR AVG & 0.007941176 \\
Canada & 2005-2007 & -0.039 & -0.028 \\
% Continue with other Canada records
Canada & AUTO-TUNE AVG & -0.044777778 & clustuneR AVG & -0.053555556 \\
\bottomrule
\end{longtable}


\end{document}

\end{document}

\section{Figure captions}

\begin{figure}[h!]
	\caption{ AUTO-TUNE scoring across candidate thresholds ranging from 0\% to
		2.5\% genetic distance. The plots represent the datasets from Seattle, Middle
		Tennessee, and Northern Alberta as described by clustuneR. The y-axis
		represents the AUTO-TUNE score, with higher scores suggesting more optimal
		thresholds. The x-axis represents the candidate thresholds. None of the three
		datasets exhibited an extreme peak of over 1.9, implying multiple thresholds
		could serve well and that a more complex decision-making process that
		includes downstream metrics such as DWH or careful inspection may be
		necessary.}\label{fig:clustuner}
\end{figure}

\begin{figure}[h!]
	\caption{ AUTO-TUNE scoring across candidate thresholds from 0\% to 2.5\%
		genetic distance for eight datasets from various studies that have previously
		employed HIV-TRACE with qualitatively defined thresholds. Each plot
		represents one dataset, and the y-axis shows the AUTO-TUNE score. Higher
		scores indicate more optimal thresholds for clustering. The x-axis represents
		the range of candidate thresholds. Each plot is labeled by the respective
		studies' PubMed ID. Cases such as Dalai et al. and Bbosa et al., exemplify
		the potential for substantial score variation even within very narrow distance
		intervals, underscoring the value of a more granular and systematic approach to
		threshold selection. }\label{fig:paperComparison} \end{figure}

\begin{figure}[h!]
	\caption{ Comparative visualizations of HIV-1 CRF07\_BC networks at different
		thresholds, colored by risk factor: heterosexual (green), person who injects
		drugs (light blue), and men who have sex with men (dark blue). Panel A
		represents the network at a 1.5\% threshold, encompassing 5923 nodes. Panel B
		illustrates the network at the 0.76\% threshold, as indicated by the highest
		AUTO-TUNE score, capturing 3537 nodes. Panel C displays the network at a
		0.19\% threshold, corresponding to the second highest AUTO-TUNE score,
		comprising 1654 nodes. Panel D features the AUTO-TUNE score plot, spanning
		from 0\% to 0.5\% thresholds, with significant peaks at 0.76\% and 0.19\%
	}\label{fig:chinaScorePlot}
\end{figure}

\begin{figure}[h!]
	\caption{ Comparison of HIV networks before and after AUTO-TUNE optimization.
		The left panel shows the network constructed using HIV-TRACE's default
		threshold of 1.5\%, while the right panel displays the network after applying
		AUTO-TUNE's optimal threshold of 1.699\%. Notably, the largest cluster,
		primarily consisting of samples from China, Thailand, and Vietnam, is
		reinforced with more nodes in the AUTO-TUNEd network. Despite the increase in
		threshold, overall homophily among other countries remains consistent.
	}\label{fig:scoreAbove2}
\end{figure}

\begin{figure}[h!]
	\caption{ (A) Box plot representing the AUTO-TUNE scores across ten random
		samples at 25\%, 50\%, and 75\% of the \citep{rhee_national_2019} dataset,
		showing a trend of increasing confidence in score estimates with denser
		sampling. (B) Box plot of the selected distance thresholds across the same
		random samples at 25\%, 50\%, and 75\% proportions, demonstrating improved
		consistency in threshold selection with increased sample size. (C)
		Scatterplot of the chosen thresholds (Y-axis) against their corresponding
		AUTO-TUNE scores (X-axis) for the three subsample proportions.
	}\label{fig:subsampling}
\end{figure}

\end{document}
